### PR TITLE
feat: Add value for clusterDomain

### PR DESCRIPTION
**Pull request by alexalex89**
## Description
Add a new parameter global. clusterDomain to be able to overwrite the Kubernetes clusterDomain in Helm context.

#### PR fixes following issue(s)
Fixes # (issue number)

#### Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

#### How Has This Been Tested?
- [x] Manual

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Documentation**
	- Updated the Helm chart README to include the new `global.clusterDomain` parameter with a default value of `"cluster.local"`.

### CODE REVIEW

**Comment by alexalex89** [comment]
@sharat87 @pratapaprasanna
Please review :-)

File: deploy/helm/templates/configMap.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- $clusterDomain := .Values.global.clusterDomain -}}
```
Comment (sharat87) [inline deploy/helm/templates/configMap.yaml:1]: Can we include the default value here itself? Like

```gotemplate
{{- $clusterDomain := .Values.global.clusterDomain | default "cluster.local" -}}
```

File: deploy/helm/templates/statefulset.yaml [hunk 2]
```diff
@@ -51,7 +51,7 @@ spec:
       {{- else }}
         image: "alpine"
       {{- end }}
-        command: ['sh', '-c', "apk add redis; until redis-cli -h {{ include "appsmith.fullname" . }}-redis-master.{{.Release.Namespace}}.svc.cluster.local ping; do echo waiting for redis; sleep 2; done"]
+        command: ['sh', '-c', "apk add redis; until redis-cli -h {{ include "appsmith.fullname" . }}-redis-master.{{.Release.Namespace}}.svc.{{ .Values.global.clusterDomain }} ping; do echo waiting for redis; sleep 2; done"]
```
Comment (sharat87) [inline deploy/helm/templates/statefulset.yaml:54]: Let's add default here as well.

File: deploy/helm/templates/statefulset.yaml [hunk 3]
```diff
@@ -60,7 +60,7 @@ spec:
       {{- else }}
         image:  "docker.io/bitnami/mongodb:4.4.11-debian-10-r12"
       {{- end }}
-        command: ['sh', '-c', "until mongo --host appsmith-mongodb.{{.Release.Namespace}}.svc.cluster.local --eval 'db.runCommand({ping:1})' ; do echo waiting for mongo; sleep 2; done"]
+        command: ['sh', '-c', "until mongo --host appsmith-mongodb.{{.Release.Namespace}}.svc.{{ .Values.global.clusterDomain }} --eval 'db.runCommand({ping:1})' ; do echo waiting for mongo; sleep 2; done"]
```
Comment (sharat87) [inline deploy/helm/templates/statefulset.yaml:63]: Here as well please. 🙂

**Review by sharat87 — CHANGES_REQUESTED** [review]
Thank you so much for your contribution. I'm adding a small change request here, and after that we should be ready to merge this.

**Comment by alexalex89** [comment]
Thanks for your review!
I implemented the requested changes. :+1:

**Comment by sharat87** [comment]
@pratapaprasanna, can you review this when you get a chance please? Thanks!

**Comment by alexalex89** [comment]
@pratapaprasanna @sharat87 Would be great if a review could be done before the PR is closed. Thanks in advance!

**Review by pratapaprasanna — APPROVED** [review]

**Comment by pratapaprasanna** [comment]
@alexalex89  can you rebase and push the changes ?

**Comment by alexalex89** [comment]
> @alexalex89 can you rebase and push the changes ?

Hey, I did a rebase on release :-)

**Comment by alexalex89** [comment]
@pratapaprasanna Could you please merge my PR? Thanks!

**Comment by alexalex89** [comment]
@pratapaprasanna What's the current status here? Anything I could do?

**Pull request closed without merging**